libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/api/MapboxManeuverApi.kt

```diff
@@ -374,12 +376,9 @@ class MapboxManeuverApi internal constructor(
         styleId: String?,
         accessToken: String?,
         maneuvers: List<Maneuver>,
-        shieldCallback: RoadShieldsCallback
+        shieldCallback: RouteShieldCallback
     ) {
         mainJobController.scope.launch {
-            val shieldMap = hashMapOf<String, List<RoadShield>>()
-            val errorMap = hashMapOf<String, List<RoadShieldError>>()
```

Comment:
We can't make this changes here, we'd have one callback firing multiple times without an association between maneuver ID and a shield, wouldn't we? A single callback should only fire once with all the data necessary.

Reply:
Agreed, but if we look at it from a different perspective this callback is only firing once per primary, secondary and sub maneuver. Once the receiver receives the callback and if you would look at the `MapboxManeuverView.maneuverToRoadShields`, it's not a `map` anymore but a `set`. I did an exercise by removing the whole concept of `maneuverID` from `PrimaryManeuver`, `SecondaryManeuver` and `SubManeuver` and the solution works by drawing shields associated with each maneuver. So as far as the concern about associating maneuverId with a shield, we don't need that association.

Reply:
The problem for me is with a concept of a callback that fires multiple times.

If I'm registering a callback for a certain action, I expect it to fire once when the job is complete. For continuous updates we have an observer pattern. The current implementation falls somewhere in-between, developers don't know how many more times the callback can fire until the job is complete.

Reply:
Okay I have changed this a bit.
@LukasPaczos